ethereum/contracts/BasicOutboundChannel.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "./OutboundChannel.sol";
+
+// BasicOutboundChannel is a basic channel that just sends messages with a nonce.
+contract BasicOutboundChannel is OutboundChannel {
+
+    constructor() {
+        nonce = 0;
+    }
+
+    /**
+     * @dev Sends a message across the channel
+     */
+    function submit(bytes memory payload)
+        public
+        override
+    {
+        emit Message(msg.sender, nonce, payload);
+        nonce = nonce + 1;
```

Comment:
Minor point, but for consistency: Seems that the nonce here tracks the last consumed one, while in the inbound channel the nonce tracks the next free one. I think next free one is the idiomatic way, right?

Reply:
Good point! If that's the idiom then I'm onboard. Though I've just got everything working end-to-end, so am rather keen to address this in a future PR. Added https://github.com/Snowfork/polkadot-ethereum/issues/241 to track the issue.